OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs

```diff
@@ -131,8 +133,26 @@ static void UpdateNeighbours(Actor self)
 
 		public void AddedToWorld(Actor self)
 		{
-			DefaultAnimation.PlayFetchIndex(NormalizeSequence(self, Info.Sequence), () => adjacent);
-			UpdateNeighbours(self);
+			var building = self.Trait<Building>();
+
+			// If there is no "make" animation to play, then render the wall sprite directly.
+			if (building.SkipMakeAnimation || !self.Info.HasTraitInfo<WithMakeAnimationInfo>())
+			{
+				DefaultAnimation.PlayFetchIndex(NormalizeSequence(self, Info.Sequence), () => adjacent);
+				UpdateNeighbours(self);
+			}
+			else
+			{
+				// Wall adjacency rendering has lower priority.
+				// Wait for the "make" animation to finish first instead of overriding DefautAnimation.
+				self.QueueActivity(Util.SequenceActivities(
+					new WaitFor(() => building.BuildComplete),
+					new CallFunc(() =>
```

Comment:
Yes, the one one `WithMakeAnimation`. I can't introduce such an ugly dependency. :X